appservice/src/createAppService/SiteCreateStep.ts

```diff
@@ -18,21 +18,21 @@ import { AppKind, WebsiteOS } from './AppKind';
 import { IAppServiceWizardContext } from './IAppServiceWizardContext';
 
 export interface IAppSettingsContext {
-    storageConnectionString: string;
-    fileShareName: string;
```

Comment:
This feels like a breaking change so update the version accordingly. It might not change functionally, but it'll at least cause a build break

Reply:
I'm purely asking just for edification since I'll have to bump up the version if createSiteAppSettings is enforced, but I thought that making properties optional would prevent changes from being broken.  How come that would break the build?

Reply:
Because `string | undefined` can't be assigned to `string`. I don't actually know if it'll break the build and the fix would just be to use `nonNullProp` if it does - but it's more of an "on principal" thing. `IAppSettingsContext` used to guarantee it would give `createSiteAppSettings` a `fileShareName` and `storageConnectionString`, and now it no longer makes that guarantee - that seems breaking.